SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ExpectedViewDelta.cpp

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "ComponentTestUtils.h"
+#include "SpatialView/ViewDelta.h"
+#include "Tests/SpatialView/ExpectedEntityDelta.h"
+#include "Tests/SpatialView/ExptectedViewDelta.h"
+
+using namespace SpatialGDK;
+
+ExpectedViewDelta& ExpectedViewDelta::AddEntityDelta(const Worker_EntityId EntityId, const EntityChangeType ChangeType)
+{
+	EntityDeltas.Add(EntityId, { EntityId, ChangeType == ADD, ChangeType == REMOVE });
+	return *this;
+}
+
+ExpectedViewDelta& ExpectedViewDelta::AddComponentAdded(const Worker_EntityId EntityId, ComponentData Data)
```

Comment:
yeah, we kinda expect you use it in the "right way" which is probably not great. will add some assertions

Reply:
as sami just reminded me: it will fail loudly already because if the entity delta is not in the map, Unreal will throw an exception instead of creating a new element